client/components/domains/registrant-extra-info/uk-form.jsx

```diff
@@ -0,0 +1,243 @@
+				'A trading name is required for this registrant type.'
+			),
+		};
+	}
+
+	componentWillMount() {
+		// Add defaults to redux state to make accepting default values work.
+		const neededRequiredDetails = difference(
+			[ 'registrantType' ],
+			keys( this.props.contactDetailsExtra )
+		);
+
+		// Bail early as we already have the details from a previous purchase.
+		if ( isEmpty( neededRequiredDetails ) ) {
+			return;
+		}
+
+		this.props.updateContactDetailsCache( {
+			extra: pick( defaultValues, neededRequiredDetails ),
+		} );
+	}
+
+	handleChangeEvent = event => {
+		this.props.updateContactDetailsCache( {
+			extra: { [ camelCase( event.target.id ) ]: event.target.value },
+		} );
+	};
+
+	isTradingNameRequired( registrantType ) {
+		return includes(
+			[ 'LTD', 'PLC', 'LLP', 'IP', 'RCHAR', 'FCORP', 'OTHER', 'FOTHER', 'STRA' ],
+			registrantType
+		);
+	}
+
+	isRegistrationNumberRequired( registrantType ) {
+		return includes( [ 'LTD', 'PLC', 'LLP', 'IP', 'SCH', 'RCHAR' ], registrantType );
+	}
+
+	renderTradingNameField() {
+		const { contactDetails, translate } = this.props;
+		const { tradingName } = contactDetails.extra || {};
+		const tradingNameErrors = get( this.props.validationErrors, [ 'extra', 'tradingName' ], [] );
+		const isError = ! isEmpty( tradingNameErrors );
+
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel className="registrant-extra-info__trading-name" htmlFor="trading-name">
+						{ translate( 'Trading Name' ) }
+					</FormLabel>
+					<FormTextInput
+						id="trading-name"
+						value={ tradingName }
+						autoCapitalize="off"
+						autoComplete="off"
+						autoCorrect="off"
+						placeholder={ '' }
+						onChange={ this.handleChangeEvent }
+						isError={ isError }
+					/>
+					{ map( tradingNameErrors, this.renderValidationError ) }
+				</FormFieldset>
+			</div>
+		);
+	}
+
+	renderRegistrationNumberField() {
+		const { contactDetails, translate } = this.props;
+		const { registrationNumber } = contactDetails.extra || '';
+		const registrationNumberErrors = get(
+			this.props.validationErrors,
+			[ 'extra', 'registrationNumber' ],
+			[]
+		);
+
+		const isError = ! isEmpty( registrationNumberErrors );
+
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel
+						className="registrant-extra-info__registration-number"
+						htmlFor="registration-number"
+					>
+						{ translate( 'Registration Number' ) }
+					</FormLabel>
+					<FormTextInput
+						id="registration-number"
+						value={ registrationNumber }
+						autoCapitalize="off"
+						autoComplete="off"
+						autoCorrect="off"
+						placeholder={ '' }
+						onChange={ this.handleChangeEvent }
+						isError={ isError }
+					/>
+					{ map( registrationNumberErrors, this.renderValidationError ) }
+				</FormFieldset>
+			</div>
+		);
+	}
+
+	errorMessageFromCode( errorCode ) {
+		return (
+			this.errorMessages[ errorCode ] ||
+			this.props.translate( 'There was a problem with this field.' )
```

Comment:
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'There was a problem with this field' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).